source/developers/plugins/site-plugins.rst

```diff
@@ -96,59 +115,218 @@ the file:
 | ``delivery/scripts/rest/*``              | ``/scripts/rest/<plugin id path>/*``                          |
 +------------------------------------------+---------------------------------------------------------------+
 
-----------
-Publishing
-----------
+^^^^^^^^^^^^^^^^^^
+Create your plugin
+^^^^^^^^^^^^^^^^^^
+
+To create a plugin, a descriptor file  ``craftercms-plugin.yaml`` is required. Below is an example site plugin descriptor file.
+
+  .. code-block:: yaml
+     :linenos:
+     :caption: *Example craftercms-plugin.yaml file for a site plugin*
+
+     # This file describes a plugin for use in Crafter CMS
+
+     # The version of the format for this file
+     descriptorVersion: 2
+
+     # Describe the plugin
+     plugin:
+       type: site
+       id: org.craftercms.plugin.test
+       name: Site Plugin Example
+       tags:
+         - test
+       version:
+         major: 3
+         minor: 0
+         patch: 1
+       description: A simple example for site plugins
+       website:
+         name: Site Plugin Example
+         url: https://github.com/craftercms/site-plugins-example
+       media:
+         screenshots:
+           - title: Crafter CMS
+             description: Crafter CMS Example Plugin
+             url: "https://raw.githubusercontent.com/craftercms/site-plugin-example/master/.crafter/logo.svg"
+       developer:
+         company:
+           name: Crafter Software
+           email: info@craftersoftware.com
+           url: https://craftersoftware.com
+       build:
+         id: c3d2a5444e6a24b5e0481d6ba87901d0b02716c8
+         date: 2019-03-18T00:00:00Z
+       license:
+         name: MIT
+         url: https://opensource.org/licenses/MIT
+       crafterCmsVersions:
+         - major: 4
+           minor: 0
+           patch: 0
+       crafterCmsEditions:
+         - community
+         - enterprise
+
```

Comment:
Are we missing the part where we define the wiring into existing configuration files to make the Site Plugin active? For example, wire into the Sidebar?

Reply:
This depends on this ticket https://github.com/craftercms/craftercms/issues/4627

Perhaps we don't do it now, but we have to do it as part of that ticket.